### PR TITLE
Add double-click locking of draw mode

### PR DESCRIPTION
**Pull request by brianhatchl**
refs #5874

### CODE REVIEW

File: modules/ui/modes.js [hunk 1]
```diff
@@ -76,6 +77,27 @@ export function uiModes(context) {
 
 
         var debouncedUpdate = _debounce(update, 500, { leading: true, trailing: true });
+        var debouncedClick = _debounce(clickModeButton, 300);
```
Comment (brianhatchl) [inline modules/ui/modes.js]: @quincylvania Not sure if a debounced function is the best way to support a click and a dblclick on an element, but it works for my click rate.

**Comment by quincylvania** [comment]
Thanks, @brianhatchl, I'll take a look!

**Comment by quincylvania** [comment]
So at the moment we're targeting #5960 which will actually hide the mode buttons as soon as you enter the drawing mode. Double-clicking wouldn't be a viable interaction at that point, particularly for unlocking the mode. 

Alternatively we could have a lock button in the contextual UI. This is far more discoverable anyway and would work better on touchscreens.

screen shot 2019-02-27 at 4 17 59 pm

**Comment by bhousel** [comment]
Closing here, as @quincylvania said we're currently heading towards a contextual secondary menu that will include a lock button.

**Pull request closed without merging**